beacon-chain/core/blocks/block_operations.go

```diff
@@ -378,6 +378,98 @@ func isSurroundVote(data1 *pb.AttestationData, data2 *pb.AttestationData) bool {
 	return sourceEpoch1 < sourceEpoch2 && targetEpoch2 < targetEpoch1
 }
 
+/* spev-v0.6
+def process_attestation(state: BeaconState, attestation: Attestation) -> None:
+    """
+    Process ``Attestation`` operation.
+    """
+    data = attestation.data
+    attestation_slot = get_attestation_data_slot(state, data)
+    assert attestation_slot + MIN_ATTESTATION_INCLUSION_DELAY <= state.slot <= attestation_slot + SLOTS_PER_EPOCH
+
+    pending_attestation = PendingAttestation(
+        data=data,
+        aggregation_bitfield=attestation.aggregation_bitfield,
+        inclusion_delay=state.slot - attestation_slot,
+        proposer_index=get_beacon_proposer_index(state),
+    )
+
+    assert data.target_epoch in (get_previous_epoch(state), get_current_epoch(state))
+    if data.target_epoch == get_current_epoch(state):
+        ffg_data = (state.current_justified_epoch, state.current_justified_root, get_current_epoch(state))
+        parent_crosslink = state.current_crosslinks[data.crosslink.shard]
+        state.current_epoch_attestations.append(pending_attestation)
+    else:
+        ffg_data = (state.previous_justified_epoch, state.previous_justified_root, get_previous_epoch(state))
+        parent_crosslink = state.previous_crosslinks[data.crosslink.shard]
+        state.previous_epoch_attestations.append(pending_attestation)
+
+    # Check FFG data, crosslink data, and signature
+    assert ffg_data == (data.source_epoch, data.source_root, data.target_epoch)
+    assert data.crosslink.epoch == min(data.target_epoch, parent_crosslink.epoch + MAX_EPOCHS_PER_CROSSLINK)
+    assert data.crosslink.parent_root == hash_tree_root(parent_crosslink)
+    assert data.crosslink.data_root == ZERO_HASH  # [to be removed in phase 1]
+    validate_indexed_attestation(state, convert_to_indexed(state, attestation))
+*/
+
+func ProcessBlockAttestations(state *pb.BeaconState, attestation *pb.Attestation) (*pb.BeaconState, error) {
```

Comment:
Why are we implementing this? The PR says to implement `get_attestation_data_slot` not `process_attestation`